packages/mobile/src/identity/feelessVerification.ts

```diff
@@ -0,0 +1,1088 @@
+    yield call(fetchKomenciSessionState, komenciKit, e164Number)
+  }
+}
+
+// Checks if phoneHash is in cache, if it's not it asks Komenci to get it,
+// puts it in the cache, then checks again
+function* fetchPhoneHashDetails(komenciKit: KomenciKit, e164Number: string) {
+  Logger.debug(TAG, '@fetchPhoneHashDetails', 'Starting fetch')
+  try {
+    yield call(fetchPhoneHashDetailsFromCache, e164Number)
+  } catch (error) {
+    const feelessVerificationState: FeelessVerificationState = yield select(
+      feelessVerificationStateSelector
+    )
+
+    const blsBlindingClient = new ReactBlsBlindingClient(networkConfig.odisPubKey)
+    const pepperQueryResult: Result<GetDistributedBlindedPepperResp, FetchError> = yield call(
+      [komenciKit, komenciKit.getDistributedBlindedPepper],
+      e164Number,
+      DeviceInfo.getVersion(),
+      blsBlindingClient
+    )
+
+    if (!pepperQueryResult.ok) {
+      Logger.debug(TAG, '@fetchPhoneHashDetails', 'Unable to query for pepper')
+      throw pepperQueryResult.error
+    }
+
+    yield put(
+      feelessUpdateVerificationState({
+        ...feelessVerificationState,
+        komenci: {
+          ...feelessVerificationState.komenci,
+          pepperFetchedByKomenci: true,
+        },
+      })
+    )
+
+    yield put(updateE164PhoneNumberSalts({ [e164Number]: pepperQueryResult.result.pepper }))
+    yield call(fetchPhoneHashDetailsFromCache, e164Number)
+  }
+}
+
+function* fetchOrDeployMtw(
+  contractKit: ContractKit,
+  komenciKit: KomenciKit,
+  walletAddress: string,
+  e164Number: string
+) {
+  Logger.debug(TAG, '@fetchOrDeployMtw', 'Starting fetch')
+  let feelessVerificationState: FeelessVerificationState = yield select(
+    feelessVerificationStateSelector
+  )
+  const storedUnverifiedMtwAddress = feelessVerificationState.komenci.unverifiedMtwAddress
+  let deployedUnverifiedMtwAddress: string | null = null
+  let komenciError: FetchError | TxError | InvalidWallet | undefined
+
+  // If there isn't a MTW stored for this session, ask Komenci to deploy one
+  if (!storedUnverifiedMtwAddress) {
+    // This try/catch block is a workaround because Komenci will throw an error
+    // if a wallet was already deployed in a session. This is only fatal if
+    // we can't recover the MTW address or there is no quota left on the session
+    try {
+      const deployWalletResult: Result<string, FetchError | TxError | InvalidWallet> = yield call(
+        [komenciKit, komenciKit.deployWallet],
+        CURRENT_MTW_IMPLEMENTATION_ADDRESS
+      )
+
+      if (!deployWalletResult.ok) {
+        Logger.debug(TAG, '@fetchOrDeployMtw', 'Unable to deploy MTW')
+        throw deployWalletResult.error
+      }
+
+      deployedUnverifiedMtwAddress = deployWalletResult.result
+    } catch (error) {
+      // Fetch session state now that there should be no cases where we don't have a  MTW
+      yield call(fetchKomenciSessionState, komenciKit, e164Number)
```

Comment:
IIUC, this can result in receiving an MTW address from Komenci, in which case we want to continue without error, using that address. I'm guessing that was unintentionally removed and the `deployedUnverifiedMtwAddress` should be updated here.

Reply:
Spot on. Thanks!